Snowflake.Data/Client/SnowflakeDbConnection.cs

```diff
@@ -24,8 +26,155 @@ public class SnowflakeDbConnection : DbConnection
 
         internal int _connectionTimeout;
 
+        private bool isActive = false;
+
         private bool disposed = false;
 
+        /// <summary>
+        ///     Connection pool 
+        /// </summary>
+        private static ConcurrentDictionary<Tuple<string, SecureString>, SnowflakeDbConnection> connectionPool;
+        private static int minPoolSize;
+        private static int maxPoolSize;
+        private static int cleanupCounter;
+        private static int currentCounter;
+        private const int MIN_POOL_SIZE = 0;
+        private const int MAX_POOL_SIZE = 10;
+        private const int CLEANUP_COUNTER = 0;
+
+        private static void initConnectionPool()
+        {
+            connectionPool = new ConcurrentDictionary<Tuple<string, SecureString>, SnowflakeDbConnection>();
+            minPoolSize = MIN_POOL_SIZE;
+            maxPoolSize = MAX_POOL_SIZE;
+            cleanupCounter = CLEANUP_COUNTER;
+            currentCounter = 0;
+        }
+
+        private static void cleanNonActiveConnections()
+        {
+            if (null == connectionPool)
+            {
+                initConnectionPool();
+            }
+
+            List<Tuple<string, SecureString>> keys = new List<Tuple<string, SecureString>>(connectionPool.Keys);
+            int curSize = keys.Count;
+            foreach (var item in keys)
+            {
+                SnowflakeDbConnection conn;
+                connectionPool.TryGetValue(item, out conn);
+                if (!conn.isActive && curSize > minPoolSize)
+                {
+                    connectionPool.TryRemove(item, out conn);
+                    conn.CloseConnection();
+                    conn.Dispose(false);
+                    curSize--;
+                }
+            }
+        }
+
+        private static SnowflakeDbConnection getConnection(string connStr, SecureString pw)
+        {
+            var connKey = Tuple.Create(connStr, pw);

```

Comment:
The password might not need to be part of the key.

Reply:
I have seen systems where multiple users could have the same username, but with different passwords. 

Snowflake better not be doing that.